### ISSUE TITLE
linarith can no longer solve a simple problem

### ISSUE THREAD

**Issue by eric-wieser** [leanprover-community/mathlib4#8875]
Adapted [from zulip](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/linarith.20regression/near/406082084):

> This works in mathlib3:
> ```lean
> import data.real.basic
> 
> example (a b c d e : ℚ)
>     (ha : 2 * a + b + c + d + e = 4)
>     (hb : a + 2 * b + c + d + e = 5)
>     (hc : a + b + 2 * c + d + e = 6)
>     (hd : a + b + c + 2 * d + e = 7)
>     (he : a + b + c + d + 2 * e = 8) :
>     e = 3 := by
>   linarith
> ```
> but no longer works in mathlib4:
> ```lean
> import Mathlib
> 
> example (a b c d e : ℚ)
>     (ha : 2 * a + b + c + d + e = 4)
>     (hb : a + 2 * b + c + d + e = 5)
>     (hc : a + b + 2 * c + d + e = 6)
>     (hd : a + b + c + 2 * d + e = 7)
>     (he : a + b + c + d + 2 * e = 8) :
>     e = 3 := by
>   linarith (config := { oracle := .fourierMotzkin })
> ```

Possibly related to #2717

**Comment by dwrensha** [comment]
The example works now, probably thanks to #12014.

**Comment by eric-wieser** [comment]
Indeed; I've edited the test-case to avoid the effects of that PR. The bug is still present, but it's now opt-in.

**Comment by vihdzp** [comment]
Or perhaps, it just hid some error in the `RBTree` implementation.
